packages/vite/src/node/config.ts

```diff
@@ -369,6 +370,13 @@ export async function resolveConfig(
     }
   }
 
+  const server = config.server || {}
+  const serverRoot = path.posix.resolve(
+    resolvedRoot,
+    server.fsServe?.root || searchForWorkspaceRoot(resolvedRoot)
+  )
+  server.fsServe = { root: serverRoot }
```

Comment:
Should we extract this part to `resolveServerOptions` similar to `resolveBuildOptions` so it's more clear of its propose?